projects/admin/src/app/record/search-view/document-advanced-search-form/document-advanced-search-form.component.ts

```diff
@@ -0,0 +1,324 @@
+          this.model = queryModel;
+        } else {
+          this.localeStorage.remove(DocumentAdvancedSearchFormComponent.LOCALE_STORAGE_NAME);
+        }
+      }
+    }
+  }
+
+  /**
+   * Init field
+   * @param field The current field
+   * @param fieldParentKey The key of parent field
+   * @param subject - Behavior Event
+   */
+  private initField(field: any, fieldParentKey: string, subject: BehaviorSubject<any>): void {
+    if (Object.keys(this.fieldDataConfig).some(key => key === fieldParentKey)) {
+      field.templateOptions = {
+        options: subject.asObservable(),
+        minItemsToDisplaySearch: 10,
+        sort: false,
+      };
+      field.type = 'select';
+      subject.next(this.fieldDataConfig[fieldParentKey]);
+    } else {
+      field.type = 'input';
+    }
+  }
+
+  /**
+   * Init field search type
+   * @param field The current field
+   * @param fieldParentKey The key of parent field
+   * @param subject - Behavior Event
+   */
+  private initFieldSearchType(field: any, fieldParentKey: string, subject: BehaviorSubject<any>): void {
+    if (Object.keys(this.fieldsSearchTypeConfig).some(key => key === fieldParentKey)) {
+      field.templateOptions = {
+        options: subject.asObservable(),
+        sort: false,
+      };
+      subject.next(this.fieldsSearchTypeConfig[fieldParentKey]);
+      const fieldSearchTypeValue = this.fieldsSearchTypeConfig[fieldParentKey][0].value;
+      if (
+        !field.formControl.value
+        || (field.formControl.value && fieldSearchTypeValue !== field.formControl.value)
+      ) {
+        field.formControl.setValue(fieldSearchTypeValue);
+      }
+    }
+  }
+
+  /**
+   * Generate query by model
+   * @param model - The form model
+   * @returns a query string
+   */
+  private generateQueryByModel(model: ISearchModel): string {
+    return this.advancedSearchService.generateQueryByModel(model);
+  }
+
+  /** Initializing the Formly configuration */
+  private initializeFieldConfig(): void {
+    this.fieldDataConfig = this.advancedSearchService.getFieldsData();
+    this.fieldsSearchTypeConfig = this.advancedSearchService.getFieldsSearchType();
+    this.fieldsConfig = [
+      {
+        fieldGroupClassName: 'row',
+        fieldGroup: [
+          {
+            fieldGroupClassName: 'row',
+            className: 'col-11',
+            fieldGroup: [
+              {
+                className: 'col-6',
+                type: 'select',
+                key: 'field',
+                defaultValue: 'title',
+                templateOptions: {
+                  sort: false,
+                  hideLabel: true,
+                  required: true,
+                  options: this.advancedSearchService.getFieldsConfig(),
+                },
+              },
+              {
+                className: 'col-2',
+                type: 'select',
+                key: 'searchType',
+                defaultValue: false,
+                templateOptions: {
```

Comment:
Is it required?

Reply:
Yes, it's necessary to have the empty options array.